lib/url.js

```diff
@@ -146,7 +148,20 @@ const {
   CHAR_COLON,
 } = require('internal/constants');
 
+let urlParseWarned = false;
+
 function urlParse(url, parseQueryString, slashesDenoteHost) {
+  if (getOptionValue('--pending-deprecation') && !urlParseWarned) {
```

Comment:
```suggestion
  if (!urlParseWarned && getOptionValue('--pending-deprecation')) {
```

to bypass the conditional more efficiently in the meantime for existing `url.parse()` users

Reply:
Thanks for the review. I just updated the pull request.